hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/rollout/rollout/GroupsLegendLayout.java

```diff
@@ -141,29 +151,45 @@ public void populateTotalTargets(Long totalTargets) {
      * Populates the legend based on a list of anonymous groups. They can't have
      * unassigned targets.
      * 
-     * @param targetsPerGroup
+     * @param listOfTargetCountPerGroup
      *            list of target counts
      */
-    public void populateGroupsLegendByTargetCounts(final List<Long> targetsPerGroup) {
+    public void populateGroupsLegendByTargetCounts(final List<Long> listOfTargetCountPerGroup) {
         loadingLabel.setVisible(false);
 
-        for (int i = 0; i < groupsLegend.getComponentCount(); i++) {
+        for (int i = 0; i < getGroupsWithoutToBeContinuedLabel(listOfTargetCountPerGroup.size()); i++) {
             final Component component = groupsLegend.getComponent(i);
             final Label label = (Label) component;
-            if (targetsPerGroup.size() > i) {
-                final Long targetCount = targetsPerGroup.get(i);
-                label.setValue(
-                        getTargetsInGroupMessage(targetCount, i18n.getMessage("textfield.rollout.group.default.name", i + 1)));
+            if (listOfTargetCountPerGroup.size() > i) {
+                final Long targetCount = listOfTargetCountPerGroup.get(i);
+                label.setValue(getTargetsInGroupMessage(targetCount,
+                        i18n.getMessage("textfield.rollout.group.default.name", i + 1)));
                 label.setVisible(true);
             } else {
                 label.setValue("");
                 label.setVisible(false);
             }
         }
 
+        showToBeContinueLabel(listOfTargetCountPerGroup);
+
         unassignedTargetsLabel.setValue("");
         unassignedTargetsLabel.setVisible(false);
+    }
 
+    private void showToBeContinueLabel(final List<?> listOfTargetCountPerGroup) {
+        if (listOfTargetCountPerGroup.size() > MAX_GROUPS_TO_BE_DISPLAYED) {
+            groupsLegend.getComponent(MAX_GROUPS_TO_BE_DISPLAYED).setVisible(true);
+        } else if (groupsLegend.getComponentCount() > listOfTargetCountPerGroup.size()) {
+            groupsLegend.getComponent(groupsLegend.getComponentCount() - 1).setVisible(false);
```

Comment:
that is unclear what you want to do here, could you extract it to additional method?